File: 7_kyu/The_Poet_And_The_Pendulum.jl
function pendulum(numbers)
    sorted = sort(numbers)
    result = similar(sorted)
    destination = (1+lastindex(result))÷2
    step = 1
    for v in sorted
        result[destination] = v
        destination += step
        step = step < 0 ? -step+1 : -step-1
    end
    result
end

File: 7_kyu/The_Poet_And_The_Pendulum.go
package kata

import "sort"

func Pendulum(values []int) []int {
	n := len(values)
	work := make([]int, n)
	// Create work copy. Source array will be unchanged.
	copy(work, values)
	// Simple sort
	sort.Ints(work)
	// Pendulum arrangement of sorted values
	pendulum := make([]int, n)
	for i := range pendulum {
		var dst int
		if i&1 == 0 {
			dst = (n - 1 - i) / 2
		} else {
			dst = (n + i) / 2
		}
		pendulum[dst] = work[i]
	}
	return pendulum
}
